module/documents/actor/HitDice.mjs

```diff
@@ -0,0 +1,172 @@
+/**
+ * Object describing the hit dice for an actor.
+ *
+ * @param {Actor5e} actor
+ */
+export default class HitDice {
+  constructor(actor) {
+    /**
+     * Store a reference to the actor.
+     * @type {Actor5e}
+     */
+    this.actor = actor;
+
+    /**
+     * Remaining hit dice.
+     * @type {number}
+     */
+    this.remaining = 0;
+
+    /**
+     * The actor's total amount of hit dice.
+     * @type {number}
+     */
+    this.total = 0;
+
+    /**
+     * All valid die sizes derived from all classes.
+     * @type {Set<number>}
+     */
+    this.sizes = new Set();
+
+    /**
+     * Store valid class items.
+     * @type {Set<Item5e>}
+     */
+    this._classes = new Set();
```

Comment:
I suppose this is marked as an underscore to indicate that it shouldn't be mutated externally? There are ways we can guarantee that, but they're all a bit slow. I think we should remove the underscore since this seems to be used publicly, and reserve the underscore for `@protected` or `@internal` fields only.